### ISSUE TITLE
Auto-generated text-area form fields should have at least 10 lines by default and should increase size as the user types in text

### ISSUE THREAD

**Issue by tisto** [plone/volto#1775]

**Comment by iFlameing** [comment]
@avimishra18 we have the Textarea widget everywhere and we want it to grow dynamically so that users can see the whole text instead of the typing one.

**Comment by avimishra18** [comment]
@iFlameing just to be sure `src/components/manage/Widgets/TextareaWidget.jsx`?

**Comment by iFlameing** [comment]
@avimishra18 yes

**Comment by MAX-786** [comment]
Is this issue still open ?
If yes, I would like to know if this(below) is the final outcome we want:
### Before (currently we have height handler at bottom right)

![Screenshot (99)](https://user-images.githubusercontent.com/99530996/211146345-f1812cec-549a-4f7d-b290-ca5be290713c.png)

### After: (height of textarea will dynamically increase by 1 line but then we'll have a scroll bar on Form field)

![Screenshot (98)](https://user-images.githubusercontent.com/99530996/211146423-fe008458-4c2f-43cd-8dcd-9f765d39a96a.png)

Thanks!

**Comment by yashodharpatel** [comment]
Hello @iFlameing, Is this issue still open? can i work on it or is somebody working on it?

**Comment by stevepiercy** [comment]
@yashodharpatel 
please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html). Deleting comments as irrelevant to the issue itself.

**Comment by Prathamk07** [comment]
I wanna attempt

**Comment by stevepiercy** [comment]
@Prathamk07 please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html).

**Comment by techaakritisha** [comment]
Hello @tisto @iFlameing   I would love to work on this issue . Please assign it to me

**Comment by stevepiercy** [comment]
@techaakritisha no, @iFlameing is already assigned, and as you can see from the long list of failed attempts, you as well as they failed to follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by tazsq** [comment]
Is this issue still available?

**Comment by stevepiercy** [comment]
@tazsq using your super power of observation, you may see that the green "Open" icon reflects its status.

Please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by tazsq** [comment]
I just read somewhere that prs from first time contributors are not accepted till second half of 2026 like what happened to that?

**Comment by tazsq** [comment]
Nvm I'm working on it

**Comment by stevepiercy** [comment]
> I just read somewhere that prs from first time contributors are not accepted till second half of 2026 like what happened to that?

Where exactly?

**Comment by tazsq** [comment]
> > I just read somewhere that prs from first time contributors are not accepted till second half of 2026 like what happened to that?
> 
> Where exactly?

![Screenshot_2025-05-08-12-31-28-713_com.github.android.png](https://github.com/user-attachments/assets/14b1db61-2ca9-4379-9b4f-729d27fdb807)


Ss from a previous pr
There you go

**Comment by stevepiercy** [comment]
"suspended its review" != "not accepted".

"Volto Team" (24 people) ⊂ "All people who can review a new pull request in Volto" (600+ people).

And if you clicked the link to Read details, you would see the same message with, of course, details.

**Comment by tazsq** [comment]
![Image](https://github.com/user-attachments/assets/9db42d7b-7b51-4723-9aea-d057f46962a4)

substituting 10 in the rows property of textareawidget doesnt work because of some css applied to textArea component by semantic-ui-react. 

Other solution that would work would be use "textareaautosize" instead of "textArea" component . additional module called "react-textarea-autosize" needs to be installed for this to work.


Also this module provides min-rows property which is not provided by the semantic ui one.

 Should I go ahead and implement this/is this acceptable??
@stevepiercy @iFlameing

**Comment by stevepiercy** [comment]
@vishuverma-1 please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by stevepiercy** [comment]
@vishuverma-1 as a courtesy, we always direct first-timers to read contributing guidelines so that they save their own time, avoid common mistakes, and point them in the right direction. It doesn't necessarily mean that you did anything wrong.
